### PR TITLE
Add v1.21.0 changelog

### PR DESCRIPTION
**Pull request by denyskon**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,482 @@ This changelog goes through all the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.21.0](https://github.com/go-gitea/gitea/releases/tag/1.21.0) - 2023-11-12
+
+* BREAKING
+  * Restrict certificate type for builtin SSH server (#26789)
+  * Refactor to use urfave/cli/v2 (#25959)
+  * Move public asset files to the proper directory (#25907)
+  * Remove commit status running and warning to align GitHub (#25839)
```
Comment (lafriks) [inline CHANGELOG.md]: Warning was added back so changelog should be changed to reflect that only running was removed

Comment (denyskon) [reply CHANGELOG.md]: I don't think it's necessary, the revert PR is part of the changelog, and the blog article only mentions running, not warning.

Comment (lafriks) [reply CHANGELOG.md]: But if you look at changelog especially as this information is in different sections this is very misleading. I would say they should be compressed into the single changelog entry with reference to both PR

Comment (denyskon) [reply CHANGELOG.md]: Okay, why not actually - done :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.21.0](https://github.com/go-gitea/gitea/releases/tag/1.21.0) - 2023-11-12
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
## [1.21.0](https://github.com/go-gitea/gitea/releases/tag/1.21.0) - 2023-11-13
```

Comment (6543) [reply CHANGELOG.md]: (if not merged in a few hours ;) )

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
...
+  * Ui correction in mobile view nav bar left aligned items. (#27046)
+  * Chroma color tweaks (#26978)
+  * Improve "language stats" UI (#26968)
+  * Add `RemoteAddress` to mirrors (#26952)
+  * Fix the missing repo count (#26942)
+  * Show queue's active worker number (#26896)
+  * Fix media description render for orgmode (#26895)
+  * Remove CSS `has` selector and improve various styles (#26891)
+  * Refactor "shortsha" (#26877)
+  * S3: log human readable error on connection failure (#26856)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Redable error on S3 storage connection failure (#26856)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
...
+  * Update snap package (#27021)
+  * Fix: typo (#27009)
+  * Move some functions to service layer (#26969)
+  * Replace `util.SliceXxx`  with `slices.Xxx` (#26958)
+  * Refactor dashboard/feed.tmpl (#26956)
+  * Move repository deletion to service layer (#26948)
+  * Improve hint when uploading a too large avatar (#26935)
+  * Extract common code to new template (#26933)
+  * Move createrepository from module to service layer (#26927)
+  * Add @chenrui333 as maintainer (#26917)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix PushEvent NullPointerException jenkinsci/github-plugin (#27203) (#27251)
+  * Update go-enry to 2.8.5 (#27215) (#27217)
+  * Add index to `issue_user.issue_id` (#27154) (#27158)
+  * Upgrading the actions/checkout@4 (#27096)
+  * Use db.WithTx for AddTeamMember to avoid ctx abuse (#27095)
+  * Drop Node.js 16 and update js dependencies (#27094)
+  * Next round of `db.DefaultContext` refactor (#27089)
+  * More refactoring of `db.DefaultContext` (#27083)
+  * Reduce usage of `db.DefaultContext` (#27073)
+  * Apply lng2020 to maintainers (#27068)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
...
+  * Show manual cron run's last time (#27544) (#27577)
+  * Avoid run change title process when the title is same (#27467) (#27558)
+  * Respect SSH.KeygenPath option when calculating ssh key fingerprints (#27536) (#27551)
+  * Add Index to `action.user_id` (#27403) (#27425)
+  * Don't use subselect in `DeleteIssuesByRepoID` (#27332) (#27408)
+  * Add protected branch name description (#27257) (#27351)
+  * More `db.DefaultContext` refactor (#27265) (#27347)
+  * Add logs for data broken of comment review (#27326) (#27345)
+  * Add Index to `comment.dependent_issue_id` (#27325) (#27340)
+  * Another round of `db.DefaultContext` refactor (#27103) (#27262)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix sticky diff header background (#27697) (#27712)
+  * Upgrade go dependencies (#27599) (#27609)
+  * Don't show Link to TOTP if not set up (#27585) (#27588)
+  * Show manual cron run's last time (#27544) (#27577)
+  * Avoid run change title process when the title is same (#27467) (#27558)
+  * Respect SSH.KeygenPath option when calculating ssh key fingerprints (#27536) (#27551)
+  * Add Index to `action.user_id` (#27403) (#27425)
+  * Don't use subselect in `DeleteIssuesByRepoID` (#27332) (#27408)
+  * Add protected branch name description (#27257) (#27351)
+  * More `db.DefaultContext` refactor (#27265) (#27347)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -4,6 +4,481 @@ This changelog goes through all the changes that have been made in each release
...
+  * Another round of `db.DefaultContext` refactor (#27103) (#27262)
+  * Fix PushEvent NullPointerException jenkinsci/github-plugin (#27203) (#27251)
+  * Update go-enry to 2.8.5 (#27215) (#27217)
+  * Add index to `issue_user.issue_id` (#27154) (#27158)
+  * Upgrading the actions/checkout@4 (#27096)
+  * Use db.WithTx for AddTeamMember to avoid ctx abuse (#27095)
+  * Drop Node.js 16 and update js dependencies (#27094)
+  * Next round of `db.DefaultContext` refactor (#27089)
+  * More refactoring of `db.DefaultContext` (#27083)
+  * Reduce usage of `db.DefaultContext` (#27073)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Reduce usage of `db.DefaultContext` (#27073) (#27083) (#27089)  (#27103) (#27262)  (#27265) (#27347)
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -4,6 +4,475 @@ This changelog goes through all the changes that have been made in each release
...
+  * Update documentation for 1.21 actions (#26317)
+  * Doc update swagger doc for POST /orgs/{org}/teams (#26155)
+  * Doc sync authentication.md to zh-cn (#26117)
+  * Doc guide the user to create the appropriate level runner (#26091)
+  * Make organization redirect warning more clear (#26077)
+  * Docusaurus-ify (#26051)
+  * Update blog links (#25843)
+  * Fix default value for LocalURL (#25426)
+  * Update `from-source.zh-cn.md` & `from-source.en-us.md` - Cross Compile Using Zig (#25194)
+* MISC
```
Comment (6543) [inline CHANGELOG.md:451]: we should decrease the misc section by removing nits

Comment (6543) [reply CHANGELOG.md:451]: and document stuff that did not go into docs section

File: CHANGELOG.md [hunk 10]
```diff
@@ -4,6 +4,475 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add API for changing Avatars (#25369)
+  * Only show 'Manage Account Links' when necessary (#25311)
+  * Improve 'Privacy' section in profile settings (#25309)
+  * Substitute variables in path names of template repos too (#25294)
+  * Use fetch to send requests to create issues/comments (#25258)
+  * Reduce unnecessary DB queries for Actions tasks (#25199)
+  * Refactor indexer (#25174)
+  * Remove duplicated functions when deleting a branch (#25128)
+  * Modify the content format of the Feishu webhook (#25106)
+  * Support changing labels of Actions runner without re-registration (#24806)
```
Comment (6543) [inline CHANGELOG.md]: enhancement?

File: CHANGELOG.md [hunk 11]
```diff
@@ -4,6 +4,475 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix `ref` for workflows triggered by `pull_request_target` (#25743)
+  * Use frontend fetch for branch dropdown component (#25719)
+  * Some less naked returns (#25682)
+  * Add unit test for `HashAvatar` (#25662)
+  * Add elapsed time on debug for slow git commands (#25642)
+  * Get latest commit statuses from database instead of git data on dashboard for repositories (#25605)
+  * Adding  branch-name copy  to clipboard branches screen. (#25596)
+  * Move some files under repo/setting (#25585)
+  * Store and use seconds for timeline time comments (#25392)
+  * Refactor: TotalTimest return seconds (#25370)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Refactor: TotalTimes return seconds (#25370)
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -4,6 +4,475 @@ This changelog goes through all the changes that have been made in each release
...
+  * Remove nonsense `<a>` for commit status check icon (#26287)
+  * Add some Wiki unit tests (#26260)
+  * Start using template context function (#26254)
+  * Fix typo in metadata (#26207)
+  * Release attachments duplicated check (#26176)
+  * Move web JSON functions to web context and simplify code (#26132)
+  * Refactor improve NoBetterThan (#26126)
+  * Improve clickable area in repo action view page (#26115)
+  * Fix minor capitalization error in string (#26100)
+  * Remove `db.DefaultContext` in `routers/` and `cmd/` (#26076)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -4,6 +4,475 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add protected branch name description (#27257) (#27351)
+  * Add logs for data broken of comment review (#27326) (#27345)
+  * Add Index to `comment.dependent_issue_id` (#27325) (#27340)
+  * Fix PushEvent NullPointerException jenkinsci/github-plugin (#27203) (#27251)
+  * Update go-enry to 2.8.5 (#27215) (#27217)
+  * Add index to `issue_user.issue_id` (#27154) (#27158)
+  * Upgrading the actions/checkout@4 (#27096)
+  * Use db.WithTx for AddTeamMember to avoid ctx abuse (#27095)
+  * Drop Node.js 16 and update js dependencies (#27094)
+  * Reduce usage of `db.DefaultContext` (#27073) (#27083) (#27089)  (#27103) (#27262)  (#27265) (#27347)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Reduce usage of `db.DefaultContext` (#27073) (#27083) (#27089)  (#27103) (#27262)  (#27265) (#27347) (#26076)
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add index to `issue_user.issue_id` (#27154) (#27158)
+  * Upgrading the actions/checkout@4 (#27096)
+  * Use db.WithTx for AddTeamMember to avoid ctx abuse (#27095)
+  * Drop Node.js 16 and update js dependencies (#27094)
+  * Reduce usage of `db.DefaultContext` (#27073) (#27083) (#27089)  (#27103) (#27262)  (#27265) (#27347) (#26076)
+  * Remove the useless function `GetUserIssueStats` and move relevant tests to `indexer_test.go` (#27067)
+  * Make SSPI auth mockable (#27036)
+  * Extract auth middleware from service (#27028)
+  * Update snap package (#27021)
+  * Fix: typo (#27009)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 15]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Cleanup repo details icons/labels (#27644) (#27654)
+  * Add locale for deleted head branch (#26296)
+  * Improve multiple strings in en-US locale (#26213)
+  * Fix broken translations for package documantion (#25742)
+  * Correct translation wrong format (#25643)
+* BUILD
+  * Dockerfile small refactor (#27757) (#27826)
+  * Fix build errors on BSD (in BSDMakefile) (#27594) (#27608)
+  * Fully replace drone with actions (#27556) (#27575)
+  * Update JS and PY dependencies (#27501) (#27518)
```
Comment (silverwind) [inline CHANGELOG.md]:
```suggestion
```

Forgot `skip-changelog`

File: CHANGELOG.md [hunk 16]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Remove fomantic `item` module (#26775)
+  * Remove redundant nil check in `WalkGitLog` (#26773)
+  * Reduce some allocations in type conversion (#26772)
+  * Feat(API): add route and implementation for creating/updating repository secret (#26766)
+  * Move `modules/mirror` to `services` (#26737)
+  * Add auth-required to config.json for Cargo http registry (#26729)
+  * Make web context initialize correctly for different cases (#26726)
+  * [Refactor] getIssueStatsChunk to move inner function into own one (#26671)
+  * Remove avatarHTML from template helpers (#26598)
+  * Update JS dependencies (#26586)
```
Comment (silverwind) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Remove nfnt/resize and oliamb/cutter (#25999)
+  * Replace gogs/cron with go-co-op/gocron (#25977)
+  * Bump github.com/golang-jwt/jwt to v5 (#25975)
+  * Fix "Flash" message usage (#25895)
+  * Improve display of Labels/Projects/Assignees sort options (#25886)
+  * Remove ref name in PR commits page (#25876)
+  * Allow the use of alternative net.Listener implementations by downstreams (#25855)
+  * Refactor "Content" for file uploading (#25851)
+  * Add error info if no user can fork the repo (#25820)
+  * Upgrade go dependencies (#25819)
```
Comment (silverwind) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Move mail notification logic to service layer (#26905)
+  * Improve DeleteByID (#26904)
+  * Extract common code to new template (#26903)
+  * Expanding documentation in queue.go (#26889)
+  * Relocate the `RSS user feed` button (#26882)
+  * Refactor `og:description` to limit the max length (#26876)
+  * Feat(API): add secret deletion functionality for repository (#26808)
+  * Replace deprecated `elliptic.Marshal` (#26800)
+  * Updating the js libraries to latest version. (#26795)
+  * Use Go 1.21 for golangci-lint (#26786)
```
Comment (silverwind) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Bump go to 1.21 (#26608)
+  * Bump xgo to go-1.21.x and node to 20 in release-version (#26589)
+  * Add template linting via djlint (#25212)
+* DOCS
+  * Remove `known issue` section in Gitea Actions Doc (#27930) (#27938)
+  * Add 1.20.5 changelog (#27404) (#27412)
+  * Remove outdated paragraphs when comparing Gitea Actions to GitHub Actions (#27119)
+  * Update brew installation documentation since gitea moved to brew core package (#27070)
+  * Actions are no longer experimental, so enable them by default (#27054)
+  * Add v1.20.4 changelog (#26974)
```
Comment (silverwind) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Enable markdownlint `no-duplicate-header` (#27500) (#27506)
+  * Enable production source maps for index.js, fix CSS sourcemaps (#27291) (#27295)
+  * Update JS and Poetry dependencies and eslint (#27200) (#27201)
+  * Use Go 1.21 and update dependencies (#26878)
+  * Bump go to 1.21 (#26608)
+  * Bump xgo to go-1.21.x and node to 20 in release-version (#26589)
+  * Add template linting via djlint (#25212)
+* DOCS
+  * Remove `known issue` section in Gitea Actions Doc (#27930) (#27938)
+  * Add 1.20.5 changelog (#27404) (#27412)
```
Comment (silverwind) [inline CHANGELOG.md]:
```suggestion
```

File: CHANGELOG.md [hunk 21]
```diff
@@ -4,6 +4,474 @@ This changelog goes through all the changes that have been made in each release
...
+  * Fix build errors on BSD (in BSDMakefile) (#27594) (#27608)
+  * Fully replace drone with actions (#27556) (#27575)
+  * Update JS and PY dependencies (#27501) (#27518)
+  * Enable markdownlint `no-duplicate-header` (#27500) (#27506)
+  * Enable production source maps for index.js, fix CSS sourcemaps (#27291) (#27295)
+  * Update JS and Poetry dependencies and eslint (#27200) (#27201)
+  * Use Go 1.21 and update dependencies (#26878)
+  * Bump go to 1.21 (#26608)
+  * Bump xgo to go-1.21.x and node to 20 in release-version (#26589)
+  * Add template linting via djlint (#25212)
```
Comment (silverwind) [inline CHANGELOG.md:426]:
```suggestion
```

**Comment by denyskon** [comment]
Updated

**Comment by 6543** [comment]
I wan't to get #[28023](https://github.com/go-gitea/gitea/pull/28023) in ... as it's sec related ...

**Review by 6543 — APPROVED** [review]

**Review by puni9869 — APPROVED** [review]

File: CHANGELOG.md [hunk 22]
```diff
@@ -4,6 +4,452 @@ This changelog goes through all the changes that have been made in each release
 without substantial changes to our git log; to see the highlights of what has
 been added to each release, please refer to the [blog](https://blog.gitea.com).
 
+## [1.21.0](https://github.com/go-gitea/gitea/releases/tag/1.21.0) - 2023-11-13
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
## [1.21.0](https://github.com/go-gitea/gitea/releases/tag/1.21.0) - 2023-11-14
```

File: CHANGELOG.md [hunk 23]
```diff
@@ -4,6 +4,452 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add Upload URL to release API (#26663)
+  * Feat(API): update and delete secret for managing organization secrets (#26660)
+  * Feat: implement organization secret creation API (#26566)
+  * Add API route to list org secrets (#26485)
+  * Set commit id when ref used explicitly (#26447)
+  * PATCH branch-protection updates check list even when checks are disabled (#26351)
+  * Add file status for API "Get a single commit from a repository" (#16205) (#25831)
+  * Add API for changing Avatars (#25369)
+* BUGFIXES
+  * fixed duplicate attachments on dump on windows (#28019) (#28031)
```
Comment (6543) [inline CHANGELOG.md]:
```suggestion
  * Fix viewing wiki commit on empty repo (#28040) (#28044)
  * Enable system users for comment.LoadPoster (#28014) (#28032)
  * Fixed duplicate attachments on dump on windows (#28019) (#28031)
```

File: CHANGELOG.md [hunk 24]
```diff
@@ -4,6 +4,453 @@ This changelog goes through all the changes that have been made in each release
...
+  * Add Upload URL to release API (#26663)
+  * Feat(API): update and delete secret for managing organization secrets (#26660)
+  * Feat: implement organization secret creation API (#26566)
+  * Add API route to list org secrets (#26485)
+  * Set commit id when ref used explicitly (#26447)
+  * PATCH branch-protection updates check list even when checks are disabled (#26351)
+  * Add file status for API "Get a single commit from a repository" (#16205) (#25831)
+  * Add API for changing Avatars (#25369)
+* BUGFIXES
+  * Fix viewing wiki commit on empty repo (#28040) (#28044)
```
Comment (6543) [inline CHANGELOG.md:72]:
```suggestion
  * Fix viewing wiki commit on empty repo (#28040) (#28044)
  * Enable system users for comment.LoadPoster (#28014) (#28032)
```

**Review by 6543 — APPROVED** [review]

**Pull request merged**